src/contrib/mlir/lowerer.cpp

```diff
@@ -526,6 +526,58 @@ namespace
         return matchSuccess();
     }
 
+    // Negative
+    REWRITER(NGNegativeOp)
+    {
+        auto loc = cast<NGNegativeOp>(op).getLoc();
+
+        auto result = pass.buildOutputDefs(op, rewriter)[0];
+        NGRAPH_CHECK(result->getType().isa<MemRefType>());
+        // Note that builder's current function is still the original function body.
+        // use getBlock to get the new block instead.
+
+        // get new operands
+        Value* lhs = operands[0];
+
+        ScopedContext scope(rewriter, loc);
+        // Views
+        MemRefView vRes(result), vLHS(lhs);
+        // Index Values
+        IndexedValue iRes(result), iLHS(lhs);
+        // Bounds Index Handles
+        auto lbs = vLHS.getLbs();
+        auto ubs = vLHS.getUbs();
+        // Loop induction vars
+        auto ivs = IndexHandle::makeIndexHandles(vLHS.rank());
+        auto pivs = IndexHandle::makeIndexHandlePointers(ivs);
+        // Steps
+        auto steps = vLHS.getSteps();
+
+        NGRAPH_CHECK(lhs->getType().isa<MemRefType>());
+        Type elemTy = lhs->getType().dyn_cast<MemRefType>().getElementType();
+
+        LoopNestBuilder(pivs, lbs, ubs, steps)([&] {
+            ValueHandle val = iLHS(ivs);
+            if (auto floatTy = elemTy.dyn_cast<FloatType>())
+            {
+                ValueHandle two = intrinsics::constant_float(llvm::APFloat(2.0f), floatTy);
+                iRes(ivs) = val - val * two;
```

Comment:
Interesting! Is it not possible to do just `- val`? I think the current implementation might not be accurate for floating point values due to intermediate rounding and floating point special values.

Reply:
-val was giving errors

Reply:
What kind of error? Maybe that op is not implemented in the EDSC and we can contribute it to MLIR. Also, maybe you could try to do something like `0 - val`. That should work and would lead to precision issues.

Reply:
I suggested 0 - val before. Std doesn't have NegOp. But maybe we can add a negation operator to EDSC that does 0 - val. 
@nbpatel , you need to define the Constant value first

Reply:
nit: no need for `instrinsics::`

Reply:
error: no match for ‘operator-’ (operand type is ‘mlir::edsc::ValueHandle’)  ...get the same error for -val and 0 - val

Reply:
The operator exists. What does your code look like. Should be
```
ValueHandel zero = constant_float(...);
iRes(ivs) = zero - vale;
```